portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -84,17 +132,29 @@ private String convertToJsonUsingGson(List comments) {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+    UserService userService = UserServiceFactory.getUserService();
```

Comment:
We only want to allow the submission of comments if the user is logged in, so we need to use UserService